src/stanc/stanc.ml

```diff
@@ -79,25 +79,43 @@ let add_file filename =
 (** ad directives from the given file. *)
 let use_file filename =
   let ast =
-    try Parse.parse_file Parser.Incremental.program filename
-    with Errors.SyntaxError err ->
-      Errors.report_syntax_error err ;
+    try 
+      match Parse.parse_file Parser.Incremental.program filename with
+      | Result.Ok ast -> ast
+      | Result.Error err ->
+          let loc = Parse.syntax_error_location err
+          and msg = Parse.syntax_error_message err in
+          Errors.report_parsing_error (msg, loc) ;
+          exit 1
+    with Errors.SyntaxError err ->	    
```

Comment:
Is this only to catch the syntax errors that are not parsing errors? If so, should the parsing errors then be removed from the syntax errors? I feel like the syntax errors and their handling could be simplified further.

Reply:
Yes, exactly. As I mentioned in the first message, we're sort of halfway between a monadic and exception based approach right now.

Reply:
And I agree about simplifying further. Ideally we want no exceptions coming from `frontend` - keep the core of the compiler pure and do all the IO in the `stanc` wrapper.

Reply:
Agreed about throwing the exceptions as far to the outside as possible. That's the only way I can imagine accumulating more than one exception. We'd still have to think about how to continue parsing after an error, but presumably we could figure that out based on what Merlin does.